core/Db/Schema/Mysql.php

```diff
@@ -514,12 +514,14 @@ public function createTable($nameWithoutPrefix, $createDefinition)
     {
         $dbSettings   = new Db\Settings();
         $charset      = $dbSettings->getUsedCharset();
+        $dynamic = $charset==='utf8mb4' ? 'ROW_FORMAT=DYNAMIC' : '';
```

Comment:
technically it could be nice to still have the row format logic in `$dbSettings->getRowFormat()` then it is more reusable but not too important.
the variable you maybe want to name `$rowformat` instead of `$dynamic`